### PR TITLE
Front containers using CSS grid

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

A replacement for Section, which uses CSS grid over flex. The grid layout algorithm’s API is much closer to our designs, making it a more optimal choice for consistent and predictable outputs. Grids also lead to a shallower DOM, which improves paint performance.

## Why?

Rather than replacing Section in one go, this new component enables adoption on Fronts without having to handle all Article edge cases. See #7146 for the reverted previous attempt 

On browsers that do no support grid, we fall back to the flow layout algorithm, setting a specific width for each breakpoint;

## Screenshots

None yet.

### CODE REVIEW

**Review by ioannakok — APPROVED** [review]
🌟

File: dotcom-rendering/src/web/components/FrontGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+`;
+
+const topBorder = css`
+	border-top-style: solid;
+`;
+
+/**
+ * # Front Grid
+ *
+ * Represents a horizontal slice of a page, also known as a container.
```
Comment (sndrs) [inline dotcom-rendering/src/web/components/FrontGrid.tsx:353]: i _think_ a container can contain multiple slices? it's been a while, but iirc container > slice[s] > facia-card[s]?

Comment (mxdvl) [reply dotcom-rendering/src/web/components/FrontGrid.tsx:353]: Oh, that’s a good point, let’s use a different word to “slice”.

**Comment by sndrs** [comment]
there may already have been discussion around this, so sorry if this is going over old ground, but if this is a container component for fronts, is it clearer to call it `src/web/components/fronts/Container.tsx` (or just `FrontsContainer.tsx`)?

File: dotcom-rendering/src/web/components/FrontContainer.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+	brand,
+	brandAlt,
+	breakpoints,
+	neutral,
+} from '@guardian/source-foundations';
+import { FrontContainer } from './FrontContainer';
+
+export default {
+	component: FrontContainer,
+	title: 'Components/FrontGrid',
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/FrontContainer.stories.tsx:12]:
```suggestion
	title: 'Components/FrontContainer',
```

**Comment by jamesgorrie** [comment]
@sndrs there's an [RFC about the folder structure](https://github.com/guardian/dotcom-rendering/issues/7174) - any input would be appreciated if you're interested!

**Pull request merged**